mesonbuild/compilers/compilers.py

```diff
@@ -335,18 +341,26 @@ def get_base_link_args(options, linker, is_shared_module):
             args += linker.get_coverage_link_args()
     except KeyError:
         pass
-    # These do not need a try...except
-    if not is_shared_module and option_enabled(linker.base_options, options, 'b_lundef'):
-        args.append('-Wl,--no-undefined')
+
     as_needed = option_enabled(linker.base_options, options, 'b_asneeded')
     bitcode = option_enabled(linker.base_options, options, 'b_bitcode')
     # Shared modules cannot be built with bitcode_bundle because
     # -bitcode_bundle is incompatible with -undefined and -bundle
     if bitcode and not is_shared_module:
-        args.append('-Wl,-bitcode_bundle')
+        args.extend(linker.bitcode_args())
     elif as_needed:
         # -Wl,-dead_strip_dylibs is incompatible with bitcode
         args.extend(linker.get_asneeded_args())
+
+    # Apple's ld (the only one that supports bitcode) does not like any
+    # -undefined arguments at all, so don't pass these when using bitcode
```

Comment:
`ccrx` does not support these arguments as well. Any way to bypass it as well?

Reply:
I'm getting this error `Linker rlink does not support allow undefined`

Reply:
Apple does support `-undefined,<foo>`, just no in conjunction with `-bitcode-bundle`.

I think we just need to implement a stub function.

Reply:
Can you try again? I think I have all of the necessary functions implemented now.